### PR TITLE
HKG: move to platform config

### PR DESCRIPTION
**Pull request by jnewb1**
<h2>*Every HKG Car*</h2>

Unless_you_re_the_mongols__4fec11f3e79c3ae34c0befa023826976

### CODE REVIEW

**Comment by jnewb1** [comment]
```
HYBRID                          : AZERA_HEV_6TH_GEN, ELANTRA_HEV_2021, IONIQ, IONIQ_HEV_2022, IONIQ_PHEV_2019, KONA_HEV, SANTA_FE_HEV_2022, SANTA_FE_PHEV_2022, SONATA_HYBRID, KIA_K5_HEV_2020, KIA_NIRO_PHEV, KIA_NIRO_PHEV_2022, KIA_NIRO_HEV_2021, KIA_OPTIMA_H, KIA_OPTIMA_H_G4_FL
EV                              : IONIQ_EV_LTD, IONIQ_EV_2020, KONA_EV, KONA_EV_2022, KONA_EV_2ND_GEN, IONIQ_5, IONIQ_6, KIA_NIRO_EV, KIA_NIRO_EV_2ND_GEN, KIA_EV6, GENESIS_GV60_EV_1ST_GEN
MANDO_RADAR                     : IONIQ_EV_LTD, SANTA_FE, SONATA, PALISADE, SONATA_HYBRID, KIA_K5_HEV_2020, KIA_NIRO_EV, KIA_NIRO_PHEV, GENESIS_G70_2020
CANFD                           : KONA_EV_2ND_GEN, STARIA_4TH_GEN, IONIQ_5, IONIQ_6, TUCSON_4TH_GEN, SANTA_CRUZ_1ST_GEN, KIA_K8_HEV_1ST_GEN, KIA_NIRO_EV_2ND_GEN, KIA_NIRO_HEV_2ND_GEN, KIA_SPORTAGE_5TH_GEN, KIA_SORENTO_4TH_GEN, KIA_SORENTO_HEV_4TH_GEN, KIA_EV6, KIA_CARNIVAL_4TH_GEN, GENESIS_GV60_EV_1ST_GEN, GENESIS_GV70_1ST_GEN, GENESIS_GV80
RADAR_SCC                       : KIA_SORENTO_4TH_GEN, KIA_SORENTO_HEV_4TH_GEN, KIA_CARNIVAL_4TH_GEN, GENESIS_GV70_1ST_GEN, GENESIS_GV80
CAMERA_SCC                      : KONA_EV_2022
CHECKSUM_CRC8                   : ELANTRA_2021, ELANTRA_HEV_2021, SANTA_FE, SANTA_FE_2022, SANTA_FE_HEV_2022, SANTA_FE_PHEV_2022, SONATA, PALISADE, SONATA_HYBRID, CUSTIN_1ST_GEN, KIA_K5_2021, KIA_K5_HEV_2020, KIA_SELTOS
CHECKSUM_6B                     : HYUNDAI_GENESIS, KIA_SORENTO
LEGACY                          : ELANTRA, ELANTRA_GT_I30, HYUNDAI_GENESIS, IONIQ_HEV_2022, IONIQ_EV_LTD, VELOSTER, KIA_OPTIMA_G4, KIA_OPTIMA_H, KIA_CEED, GENESIS_G70, GENESIS_G80
UNSUPPORTED_LONGITUDINAL        : SONATA_LF, KIA_NIRO_PHEV, KIA_OPTIMA_G4_FL, KIA_OPTIMA_H_G4_FL, KIA_SORENTO
CANFD_NO_RADAR_DISABLE          : KONA_EV_2ND_GEN, IONIQ_6
```

**Comment by jnewb1** [comment]
process replay diff is only flags, also verified the sets are identical

File: selfdrive/car/hyundai/values.py [hunk 1]
```diff
@@ -52,99 +52,46 @@ def __init__(self, CP):
...
-    "Requires a <a href=\"https://comma.ai/shop/can-fd-panda-kit\" target=\"_blank\">CAN FD panda kit</a> if not using " +
-    "comma 3X for this <a href=\"https://en.wikipedia.org/wiki/CAN_FD\" target=\"_blank\">CAN FD car</a>.",
-    Column.MODEL, shop_footnote=False)
+  # these cars have not been verified to work with longitudinal yet - radar disable, sending correct messages, etc.
+  UNSUPPORTED_LONGITUDINAL = 2**19
+
+  CANFD_NO_RADAR_DISABLE = 2**20
+
+  CLUSTER_GEARS = 2**21
+  TCU_GEARS = 2**22
```
Comment (jnewb1) [inline selfdrive/car/hyundai/values.py:94]: `2**20` is much easier to understand than `1048576`

File: selfdrive/car/hyundai/hyundaican.py [hunk 2]
```diff
@@ -33,12 +33,12 @@ def create_lkas11(packer, frame, car_fingerprint, apply_steer, steer_req,
   values["CF_Lkas_ToiFlt"] = torque_fault  # seems to allow actuation on CR_Lkas_StrToqReq
   values["CF_Lkas_MsgCount"] = frame % 0x10
 
-  if car_fingerprint in (CAR.SONATA, CAR.PALISADE, CAR.KIA_NIRO_EV, CAR.KIA_NIRO_HEV_2021, CAR.SANTA_FE,
-                         CAR.IONIQ_EV_2020, CAR.IONIQ_PHEV, CAR.KIA_SELTOS, CAR.ELANTRA_2021, CAR.GENESIS_G70_2020,
-                         CAR.ELANTRA_HEV_2021, CAR.SONATA_HYBRID, CAR.KONA_EV, CAR.KONA_HEV, CAR.KONA_EV_2022,
-                         CAR.SANTA_FE_2022, CAR.KIA_K5_2021, CAR.IONIQ_HEV_2022, CAR.SANTA_FE_HEV_2022,
-                         CAR.SANTA_FE_PHEV_2022, CAR.KIA_STINGER_2022, CAR.KIA_K5_HEV_2020, CAR.KIA_CEED,
-                         CAR.AZERA_6TH_GEN, CAR.AZERA_HEV_6TH_GEN, CAR.CUSTIN_1ST_GEN):
+  if CP.carFingerprint in (CAR.SONATA, CAR.PALISADE, CAR.KIA_NIRO_EV, CAR.KIA_NIRO_HEV_2021, CAR.SANTA_FE,
```
Comment (jnewb1) [inline selfdrive/car/hyundai/hyundaican.py:36]: todo: more flags!!!!!!!!!!!

File: selfdrive/car/hyundai/hyundaican.py [hunk 3]
```diff
@@ -57,7 +57,7 @@ def create_lkas11(packer, frame, car_fingerprint, apply_steer, steer_req,
     values["CF_Lkas_SysWarning"] = 4 if sys_warning else 0
 
   # Likely cars lacking the ability to show individual lane lines in the dash
-  elif car_fingerprint in (CAR.KIA_OPTIMA_G4, CAR.KIA_OPTIMA_G4_FL):
+  elif CP.carFingerprint in (CAR.KIA_OPTIMA_G4, CAR.KIA_OPTIMA_G4_FL):
```
Comment (jnewb1) [inline selfdrive/car/hyundai/hyundaican.py:60]: and more!!!

File: selfdrive/car/hyundai/values.py [hunk 4]
```diff
@@ -52,260 +52,478 @@ def __init__(self, CP):
...
-  CAR.ELANTRA_HEV_2021: HyundaiCarInfo("Hyundai Elantra Hybrid 2021-23", video_link="https://youtu.be/_EdYQtV52-c",
+@dataclass
+class HyundaiPlatformConfig(PlatformConfig):
+  dbc_dict: DbcDict = field(default_factory=lambda: dbc_dict("hyundai_kia_generic", None))
+
+  def init(self):
+    if self.flags & HyundaiFlags.CANFD:
+      self.dbc_dict = dbc_dict('hyundai_canfd', None)
+    if self.flags & HyundaiFlags.MANDO_RADAR:
+      self.dbc_dict = dbc_dict('hyundai_kia_generic', 'hyundai_kia_mando_front_radar_generated')
```
Comment (sshane) [inline selfdrive/car/hyundai/values.py:114]: Should make a new subclass for CANFD, default can be CAN

File: selfdrive/car/hyundai/values.py [hunk 5]
```diff
@@ -52,260 +52,478 @@ def __init__(self, CP):
...
+    HyundaiCarInfo("Genesis GV80 2023", "All", car_parts=CarParts.common([CarHarness.hyundai_m])),
+    flags=HyundaiFlags.CANFD | HyundaiFlags.RADAR_SCC
+  )
+
+
+class Footnote(Enum):
+  CANFD = CarFootnote(
+    "Requires a <a href=\"https://comma.ai/shop/can-fd-panda-kit\" target=\"_blank\">CAN FD panda kit</a> if not using " +
+    "comma 3X for this <a href=\"https://en.wikipedia.org/wiki/CAN_FD\" target=\"_blank\">CAN FD car</a>.",
+    Column.MODEL, shop_footnote=False)
```
Comment (sshane) [inline selfdrive/car/hyundai/values.py:525]: This should be at the top still

Comment (jnewb1) [reply selfdrive/car/hyundai/values.py]: done

File: selfdrive/car/hyundai/values.py (hunk 1 above)
Comment (jnewb1) [reply selfdrive/car/hyundai/values.py]: hopefully we don't have to worry about whether our system is 32 bits or 64

File: selfdrive/car/hyundai/values.py [hunk 6]
```diff
@@ -152,160 +106,424 @@ class HyundaiCarInfo(CarInfo):
...
+class HyundaiPlatformConfig(PlatformConfig):
+  dbc_dict: DbcDict = field(default_factory=lambda: dbc_dict("hyundai_kia_generic", None))
+
+  def init(self):
+    if self.flags & HyundaiFlags.MANDO_RADAR:
+      self.dbc_dict = dbc_dict('hyundai_kia_generic', 'hyundai_kia_mando_front_radar_generated')
+
+
+@dataclass
+class HyundaiCanFDPlatformConfig(HyundaiPlatformConfig):
```
Comment (sshane) [inline selfdrive/car/hyundai/values.py:123]:
```suggestion
class HyundaiCanFDPlatformConfig(HyundaiPlatformConfig):
  dbc_dict: DbcDict = field(default_factory=lambda: dbc_dict("hyundai_kia_generic", None))
```

Comment (jnewb1) [reply selfdrive/car/hyundai/values.py:123]: ye

**Review by sshane — APPROVED** [review]

File: selfdrive/car/hyundai/values.py (hunk 1 above)
Comment (sshane) [reply selfdrive/car/hyundai/values.py]: We should probably have a test warning us of this, we're pretty close to 32 bits

**Comment by sunnyhaibin** [comment]
Merge conflicts, here it comes!

**Comment by crwusiz** [comment]
radar_interface.py

from openpilot.selfdrive.car.hyundai.values import DBC

  if DBC[CP.carFingerprint]['radar'] is None:

======================
value DBC removed

**Comment by jnewb1** [comment]
> value DBC removed

*moved*: https://github.com/commaai/openpilot/blob/master/selfdrive/car/hyundai/values.py#L770C1-L770C27
*removed*: soon

**Pull request merged**